### PR TITLE
SAK-49673 sitemanage. Allow import of selected tool items

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/SAK-49673

### CODE REVIEW

**Comment by adrianfish** [comment]
Screenshot 2024-02-28 at 11 55 27

File: conversations/impl/src/main/java/org/sakaiproject/conversations/impl/ConversationsServiceImpl.java [hunk 1]
```diff
@@ -2269,10 +2270,24 @@ public String[] getEventKeys() {
...
     }
 
+    @Override
+    public List<Map<String, String>> getEntityMap(String fromContext) {
+
+        try {
+            return getTopicsForSite(fromContext).stream()
+                .map(t -> Map.of("id", t.id, "title", t.title)).collect(Collectors.toList());
+        } catch (ConversationsPermissionsException cpe) {
+            log.error("Failed to get topics for site {}: {}", fromContext, cpe.toString());
```
Comment (ern) [inline conversations/impl/src/main/java/org/sakaiproject/conversations/impl/ConversationsServiceImpl.java:2285]:
```suggestion
            log.warn("Failed to get topics for site {}: {}", fromContext, cpe.toString());
```

File: kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java [hunk 2]
```diff
@@ -7733,6 +7733,36 @@ public Map<String, String> transferCopyEntities(String fromContext, String toCon
...
+		try {
+			ContentCollection collection = getCollection(getSiteCollection(siteId));
+
+			return m_storage.getResources(collection).stream().map(r -> {
+
+					String title = r.getProperties().getProperty(ResourceProperties.PROP_DISPLAY_NAME);
+					return Map.of("id", r.getId(), "title", title);
+				}).collect(Collectors.toList());
+		} catch (Exception e) {
+			log.error("Failed to get the entity map for site {}: {}", siteId, e.toString());
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java:7749]:
```suggestion
			log.warn("Failed to get the entity map for site {}: {}", siteId, e.toString());
```

File: kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java [hunk 3]
```diff
@@ -7733,6 +7733,36 @@ public Map<String, String> transferCopyEntities(String fromContext, String toCon
...
+	}
+
+	@Override
+	public boolean hasContent(String siteId) {
+
+		try {
+			ContentCollection collection = getCollection(getSiteCollection(siteId));
+			return !m_storage.getResources(collection).isEmpty();
+		} catch (Exception e) {
+			log.error("Failed to get the entity map for site {}: {}", siteId, e.toString());
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java:7761]:
```suggestion
			log.warn("Failed to get the entity map for site {}: {}", siteId, e.toString());
```

File: chat/chat-impl/impl/src/java/org/sakaiproject/chat2/model/impl/ChatEntityProducer.java [hunk 4]
```diff
@@ -69,17 +71,19 @@
...
  *
  */
 @Slf4j
 public class ChatEntityProducer implements EntityProducer, EntityTransferrer {
    
-   @Setter @Getter private EntityManager entityManager;
-   @Setter @Getter private ChatManager chatManager;
-   @Setter private SiteService siteService;
-   @Setter private UserDirectoryService userDirectoryService;
+   @Autowired private EntityManager entityManager;
```
Comment (ern) [inline chat/chat-impl/impl/src/java/org/sakaiproject/chat2/model/impl/ChatEntityProducer.java:83]: leave this as properties in components.xml

**Comment by adrianfish** [comment]
@ern I've removed the autowired changes and downgraded those log entries to warn from error.

**Review by ern — APPROVED** [review]

**Pull request merged**